x-pack/plugins/security_solution/public/common/store/inputs/helpers.ts

```diff
@@ -49,31 +61,23 @@ export const updateInputTimerange = (
   return state;
 };
 
-export const toggleLockTimeline = (linkToId: InputsModelId, state: InputsModel): InputsModel => {
-  const linkToIdAlreadyExist = state.global.linkTo.indexOf(linkToId);
-  return {
-    ...state,
-    global: {
-      ...state.global,
-      timerange: linkToIdAlreadyExist > -1 ? state.global.timerange : state.timeline.timerange,
-      linkTo:
-        linkToIdAlreadyExist > -1
-          ? [
-              ...state.global.linkTo.slice(0, linkToIdAlreadyExist),
-              ...state.global.linkTo.slice(linkToIdAlreadyExist + 1),
-            ]
-          : [...state.global.linkTo, linkToId],
-    },
-    timeline: {
-      ...state.timeline,
-      linkTo: linkToIdAlreadyExist > -1 ? [] : ['global'],
-    },
-  };
+export const toggleLockTimeline = (state: InputsModel): InputsModel => {
+  const linkToIdAlreadyExist = state.global.linkTo.indexOf('timeline');
+  return linkToIdAlreadyExist > -1
+    ? removeInputLink(['global', 'timeline'], state)
+    : addInputLink(['global', 'timeline'], state);
+};
+
+export const toggleLockSocTrends = (state: InputsModel): InputsModel => {
+  const linkToIdAlreadyExist = state.global.linkTo.indexOf('socTrends');
+  return linkToIdAlreadyExist > -1
+    ? removeInputLink(['global', 'socTrends'], state)
+    : addInputLink(['global', 'socTrends'], state);
 };
 
 export interface UpdateQueryParams {
   id: string;
-  inputId: InputsModelId;
+  inputId: 'global' | 'timeline';
```

Comment:
Could we create an `InputsQueryModelId` type? We are doing it explicitly in many places, It will be much easier to manage the values with a dedicated type.

Reply:
yeah, from my description: In a future PR, I will [implement a better type](https://github.com/elastic/kibana/pull/139508/commits/42b56b55982dfd9c87a0d63207e7f48432016bac) for this in order to keep this PR smaller.

i originally had that in this PR but it made it 100 files and 4 teams to review